sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -142,10 +205,66 @@ namespace Azure { namespace Core {
 
   public:
     /**
-     * @brief Constructs a new context with no deadline, and no value associated.
+     * @brief Constructs a new child context with no deadline, and no value associated.
+     *
+     */
+    Context()
+        : m_contextSharedState(
+            std::make_shared<ContextSharedState>(ApplicationContext.m_contextSharedState))
+    {
+    }
+
+    /**
+     * @brief Copies a context.
+     *
```

Comment:
These methods were added because the semantics of copying and moving a context are NOT intuitive (Context objects copy by reference, not by value, even though they appear to be values).

As far as I know, there's no way of documenting a method without actually declaring the method :)